packages/backend-common/src/database/connectors/postgres.ts

```diff
@@ -60,14 +65,54 @@ export function buildPgDatabaseConfig(
   dbConfig: Config,
   overrides?: Knex.Config,
 ) {
-  return mergeDatabaseConfig(
+  const mergedDbConfig = mergeDatabaseConfig(
     dbConfig.get(),
     {
       connection: getPgConnectionConfig(dbConfig, !!overrides),
       useNullAsDefault: true,
     },
     overrides,
   );
+
+  const azureConfig = dbConfig.getOptionalConfig('auth.azure');
+  if (azureConfig) {
+    try {
+      const credential = asAzureDevOpsCredential(azureConfig.get());
```

Comment:
Hey @freben 
I just wanted to let you know that I got your idea about getting Azure credentials reused from the Integration config. But I'm struggling to find a way to get access to this level of config. 

The issue is that `DatabaseManager.ts` is getting a database-related part of the config [passed into the constructor](https://github.com/backstage/backstage/blob/master/packages/backend-common/src/database/DatabaseManager.ts#L79). 

If you have an idea how I can instantiate [ScmIntegrations](https://github.com/backstage/backstage/blob/master/packages/integration/src/ScmIntegrations.ts#L56) inside `postgres.ts` it will help me to move forward with my PR.

Reply:
Whilst you may be able to reuse some of the code from `DefaultAzureDevOpsCredentialsProvider` around converting configuration to `TokenCredential`, I would avoid coupling the `integrations.azure` yaml configuration.

The `integration.azure` configuration is used to configure Azure DevOps. Whilst that configuration may use Entra ID, it may not (i.e. via a PAT).  There is also no guarantee that a customer hosting in Azure even has an Azure Dev Ops account, let alone configured it in Backstage, so I wouldn't want to tie postgres to a Source Control integration.

I have put together a larger RFC for consolidating how different plugins configure Entra ID configuration in #21158

Reply:
I agree with @afscrome.
We have a case where we do not use Azure for Source Control integration and Postgres. So it was a way for us. But it might be a challenge for those who have combined.